staging/src/k8s.io/apiextensions-apiserver/pkg/cmd/server/options/options.go

```diff
@@ -99,6 +103,22 @@ func (o CustomResourceDefinitionsServerOptions) Config() (*apiserver.Config, err
 	return config, nil
 }
 
+// New creates a new generic API service object using this option.
+func (o CustomResourceDefinitionsServerOptions) New() (*genericapiserver.GenericAPIServer, error) {
```

Comment:
Don't put this here in the options. The logic below belong into `completeConfig.New` https://github.com/kubernetes/kubernetes/blob/2bf66c377d60276e017e78ab9dbc0490f26baa17/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/apiserver.go#L122. Why do we externalize it?

Reply:
Is it look good now that we moved those fields into config?